src/TestSuite/Constraint/EventFiredWith.php

```diff
@@ -0,0 +1,101 @@
+<?php
+namespace Cake\TestSuite\Constraint;
+
+use Cake\Event\Event;
+use Cake\Event\EventManager;
+use PHPUnit_Framework_AssertionFailedError;
+use PHPUnit_Framework_Constraint;
+
+/**
+ * EventFiredWith constraint
+ *
+ * Another glorified in_array check
+ */
+class EventFiredWith extends PHPUnit_Framework_Constraint
+{
+    /**
+     * Array of fired events
+     *
+     * @var EventManager
+     */
+    protected $_eventManager;
+
+    /**
+     * Event data key
+     *
+     * @var string
+     */
+    protected $_dataKey;
+
+    /**
+     * Event data value
+     *
+     * @var string
+     */
+    protected $_dataValue;
+
+    /**
+     * Constructor
+     *
+     * @param EventManager $eventManager Event manager to check
+     * @param string $dataKey Data key
+     * @param string $dataValue Data value
+     */
+    public function __construct($eventManager, $dataKey, $dataValue)
+    {
+        parent::__construct();
+        $this->_eventManager = $eventManager;
+        $this->_dataKey = $dataKey;
+        $this->_dataValue = $dataValue;
+    }
+
+    /**
+     * Checks if event is in fired array
+     *
+     * @param mixed $other Constraint check
+     * @return bool
+     */
+    public function matches($other)
+    {
+        $firedEvents = [];
+        $list = $this->_eventManager->getEventList();
+        $totalEvents = count($list);
+        for ($e = 0; $e < $totalEvents; $e++) {
+            $firedEvents[] = $list[$e];
```

Comment:
Should I submit a separate PR for that change or do you mind me throwing it in here? I don't mind either way, although if it's separate this PR should wait for it otherwise there'll be another one to update this block :)